app/src/main/java/com/vishaltelangre/nerdcalci/core/MathEngine.kt

```
@@ -735,4 +736,54 @@ object MathEngine {
+
+        return if (exponentInt == null) {
+            // Fallback to Double for non-integer exponents that are within magnitude limits
+            val rightDouble = exponent.toDouble()
+            val result = base.toDouble().pow(rightDouble)
+            if (result.isInfinite()) throw ArithmeticException("Calculation result is too large")
+            if (result.isNaN()) throw ArithmeticException("Undefined")
+            BigDecimal(result, mc)
+        } else {
+            try {
+                base.pow(exponentInt, mc)
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)